### DISCUSSION TITLE
How to get child Navigation nodes

### DISCUSSION THREAD

**Discussion by metkhoo** [pnp/pnpcore#1472] [Q&A] [answered]
Hi
**Question about Navigation**

The PnP Core SDK documentation https://pnp.github.io/pnpcore/using-the-sdk/navigation-intro.html states 
"Those navigation nodes can then have child nodes"

It clearly also tells how to retrieve the nav nodes at the top level of eitehr QucikLaunch or TopNavigation.  In my case I'm interested in QuickLaunch so this line works:

```
// Get Quick Launch navigation nodes
var topLevelNodes = await context.Web.Navigation.GetAsync(n => n.QuickLaunch);
```

I can then iterate around "topLevelNodes" but I want to find the Children of a node

In the PnP powershell, there is a property "navNode.Children" that allows me to access the Children

How do I gain access the child nodes of this parent node in PnP.Core?

I thought it might be 
`await topLevelNodes.LoadAsync(p => p.Children) `

But that does not work.

Any help appreciated
Thx

**Comment by jansenbe** [comment, marked as answer]
@metkhoo : below code works for me:

```csharp
var nodes = await context.Web.Navigation.GetAsync(n => n.QuickLaunch);
foreach (var item in nodes.QuickLaunch.AsRequested())
{
    if (item.Title == "I have child nodes")
    {
        var childNodes = await item.GetChildNodesAsync();
        foreach(var childNode in childNodes)
        {
            Console.WriteLine(childNode.Title);
        }
    }
}
```

**Reply by metkhoo** [reply]
Many thanks Bert!
